compiler/rustc_transmute/src/lib.rs

```diff
@@ -123,20 +124,14 @@ mod rustc {
             tcx: TyCtxt<'tcx>,
             param_env: ParamEnv<'tcx>,
             c: Const<'tcx>,
-        ) -> Option<Self> {
+        ) -> Option<Result<Self, ErrorGuaranteed>> {
```

Comment:
```suggestion
        ) -> Result<Option<Self>, ErrorGuaranteed> {
```
general convention: errors are "more important/immediate/whatever" than ambiguity